File: week2/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");
    int i = 0;
    int letter_count = 0;
    int word_count = 0;
    int sentence_count = 0;
    while (text[i] != '\0')
    {
        // Judge the text[i] and count the letter word sentence
        if (isalpha(text[i]))
        {
            letter_count++;
        }
        //space mean a word is end
        else if (isspace(text[i]))
        {
            word_count++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence_count++;
        }
        i++;
    }
    //The last word is end with symbol not space
    word_count++;

    // count the index
    float L = (float)letter_count / word_count * 100;
    float S = (float)sentence_count / word_count * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int index_round = round(index);

    //judge what is output by compare index
    if (index_round > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index_round < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index_round);
    }

}

File: week6Python/pset6/readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letter = 0
    words = 0
    sentence = 0
    for word in text:
        # Judge the word and count the letter word sentence
        if word.isalpha():
            letter += 1
        # space mean a word is end
        elif word == " ":
            words += 1
        elif word == '.' or word == '!' or word == '?':
            sentence += 1

    # The last word is end with symbol not space
    words += 1

    # count the index
    L = letter / words * 100
    S = sentence / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    index = round(index)

    # judge what is output by compare index
    if index > 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


main()
